data/url/url.js

```diff
@@ -115,7 +115,8 @@ module.exports = connect.behavior("data/url",function(baseConnect){
 			if(typeof this.url === "object") {
 
 				if(typeof this.url[reqOptions.prop] === "function"){
-					return this.url[reqOptions.prop](params);
+					let res = this.url[reqOptions.prop](params);
+					return (res instanceof Promise) ? res : new Promise((resolve, reject) => res.then(resolve).fail(reject));
```

Comment:
do you know `fail` will be available?  What if it's an A+ promise from another window?  `fail` wouldn't exist, only `catch`.